app/components/project-geometries/modes/lots.js

```diff
@@ -84,11 +84,17 @@ export default class LotsComponent extends Component {
   generateBuffer() {
     const { features } = this.get('selectedLots');
 
-    if (!features.length) return {};
+    if (!features.length) {
+      this.set('geometricProperty', {
+        type: 'FeatureCollection',
+        features: [],
+      });
+      return;
+    }
     const [{ geometry }] = features;
     const { length } = features;
 
-    return waitForProperty(this, 'hydrateFeatures.isIdle')
+    waitForProperty(this, 'hydrateFeatures.isIdle')
```

Comment:
This can definitely happen in parallel with other things but we should keep in mind that `waitForProperty` is async and we might need to await it somewhere

Reply:
does `return` give you `await` behavior?